### ISSUE TITLE
FO Application icon broken/not used for macOS app bundle

### ISSUE THREAD

**Issue by Vezzra** [freeorion/freeorion#3114]
Bug Report
==========

Environment
-----------

* **FreeOrion Version**: f812ea6d03694ee7a7c95b494e9b2eec6631b78a and later
* **Operating System**: macOS
* **Fetched as** 
  * Compiled from source

Description
-----------

The macOS application does not show/use the FO icon anymore, only the generic icon. Bisecting revealed that the issue has been introduced by f812ea6d03694ee7a7c95b494e9b2eec6631b78a. What's really strange is that although that commit happened on February 8th, I could still build FO on my system and the correct icon would be used until May.

Then, for unknown (to me) reasons, I got the generic icon. Since then, trying to build FO with commits before May, but after f812ea6d03694ee7a7c95b494e9b2eec6631b78a, also gives me an application without the FO icon. Building with commits before f812ea6d03694ee7a7c95b494e9b2eec6631b78a gives me an application with the FO icon.

@adrianbroher, f812ea6d03694ee7a7c95b494e9b2eec6631b78a is part one of your PRs - any idea what's wrong here?

Expected Result
---------------

Building FO on macOS should give an application that displays the standard FO icon, not the generic macOS app icon.

Steps to reproduce
------------------

Build FO on macOS following the standard recommended procedure.

**Comment by adrianbroher** [comment]
@Vezzra

In case it happens suddenly without related code changes I would assume that your build environment changed, e.g. you ran brew update or that you have reused some stale build cache for quite some time and now finally ran the build on a clean slate or now reinstalled the application the first time properly.

Can you verify that the Application Icon is located within the `FreeOrion.app/Resources/` directory of a newly built Application bundle and that the file is indeed a working Apple icon file?

Can you also verify that the `FreeOrion.app/Info.plist` file exists and check if its `CFBundleIconFile` entry points towards the value `FreeOrion` (file path to icon within Resources directory without extension)?

Also what version of CMake are you using (`cmake --version`) in case you may have installed a version with buggy app bundle handling?

Maybe a look into the Applue Bundle documentation helps here for identifiying further problems https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFBundles/BundleTypes/BundleTypes.html#//apple_ref/doc/uid/10000123i-CH101-SW1

**Comment by Vezzra** [comment]
> ...or that you have reused some stale build cache for quite some time...

If I had to guess, that would be the prime candidate. My Mac isn't a very potent machine, so I only do complete rebuilds when necessary.

> Can you verify that the Application Icon is located within the `FreeOrion.app/Resources/` directory of a newly built Application bundle and that the file is indeed a working Apple icon file?

Nope, the icon file is not there. Only the `default` folder containing all the content stuff, `Changelog.md` and `main.nib`.

Obviously the icon file does not get copied into the `Resource` folder when the app bundle gets created, the question is why.

> Can you also verify that the `FreeOrion.app/Info.plist` file exists and check if its `CFBundleIconFile` entry points towards the value `FreeOrion` (file path to icon within Resources directory without extension)?

The `Info.plist` file exists, but in `FreeOrion.app/Contents/` (which should be correct, directly under the `*.app` folder of app bundles only `Contents`resides, everything else must be within that folder AFAIK). The relevant entry for the icon seems to be correct:
```
	<key>CFBundleIconFile</key>
	<string>FreeOrion</string>
```

> Also what version of CMake are you using (`cmake --version`) in case you may have installed a version with buggy app bundle handling?

```
user$ cmake --version
cmake version 3.14.5
```

Looking at the changes of the f812ea6d03694ee7a7c95b494e9b2eec6631b78a commit, I notice the following: You added the `FreeOrion.icns` file as a resource to the client app bundle on the "darwin" platform, and removed it from the list of assets which just get copied to the `Resources` folder (the remaining entries there, `Changelog.md` and `default`, still get copied correctly).

The other entry which is treated as a resource for the app bundle, `main.xib`, gets processed correctly, as the presence of the `main.nib` file proves (without that the app wouldn't work at all). But the difference here is that `main.xib` does not get just copied, it needs to be compiled into the `main.nib` file that ends up in `Resources`.

The `FreeOrion.icns` file isn't a resource in that sense - it isn't something that needs to be processed/compiled into something else, it just needs to be copied. Maybe that's the problem, and just removing it from that list of "resources" and putting its handling back to the list of "assets" resolves the issue?

However, even if that works the question remains if that were the correct solution, or just a hackaround. I guess you moved handling of that resource to that other place for a reason, and that that should work. I have no idea why it doesn't however...

**Comment by robertjschulz** [comment]
i confirm this bug in the newest download.

**Comment by Vezzra** [comment]
Yes, there hasn't been any attempt to fix this yet, unfortunately...
